mongoexport/options.go

```diff
@@ -83,3 +87,61 @@ func (inputOptions *InputOptions) GetQuery() ([]byte, error) {
 	}
 	panic("GetQuery can return valid values only for query or queryFile input")
 }
+
+// Options represents all possible options that can be used to configure mongoexport.
+type Options struct {
+	*options.ToolOptions
+	*OutputFormatOptions
+	*InputOptions
+	ParsedArgs []string
+}
+
+// ParseOptions reads command line arguments and converts them into options that can be used to configure mongoexport.
+func ParseOptions(rawArgs []string) (Options, error) {
+	// initialize command-line opts
+	opts := options.New("mongoexport", Usage,
+		options.EnabledOptions{Auth: true, Connection: true, Namespace: true, URI: true})
+	outputOpts := &OutputFormatOptions{}
+	opts.AddOptions(outputOpts)
+	inputOpts := &InputOptions{}
+	opts.AddOptions(inputOpts)
+	opts.AddKnownURIParameters(options.KnownURIOptionsReadPreference)
```

Comment:
This is part of the fix for https://jira.mongodb.org/browse/TOOLS-2217. Adding a known URI param will stop it from logging readPreference as an unsupported param, but it doesn't fix the fact that read pref isn't set on the ToolOptions. I think to get that behavior, common will need to be changed to have a method like NewMongoWriteConcernOpts for read preferences to take both a read pref string and a connstring.

Reply:
I added a comment to TOOLS-2217 to track this.

Reply:
I'd say go ahead and implement it to unblock this PR. (@xdg is it okay to bump the priority on TOOLS-2217?)

Reply:
Sure.

Reply:
I'll put TOOLS-2217 in progress and block 1836 on it then.